pkg/resource/training_job/custom_update.go

```diff
@@ -56,10 +59,41 @@ func (rm *resourceManager) buildProfilerRuleConfigUpdateInput(desired *resource,
 			}
 		}
 	}
+	// If the length of this slice is zero that only the contents of the profile rule have changed
+	if len(profilerRuleInput) == 0 {
+		return errors.New("cannot modify a profiler rule.")
+	}
 	input.SetProfilerRuleConfigurations(profilerRuleInput)
 	return nil
 }
 
+// markNonUpdatableRules returns a map containing the rules that are not eligible for update.
+// In addition it returns an error if a rule gets removed.
+func markNonUpdatableRules(profilerRuleDesired []*svcapitypes.ProfilerRuleConfiguration, profilerRuleLatest []*svcapitypes.ProfilerRuleConfiguration) (map[string]int, error) {
+	commonRulesMap := map[string]int{}
+	latestRulesMap := map[string]int{}
+	for _, rule := range profilerRuleLatest {
+		if ackcompare.IsNotNil(rule.RuleConfigurationName) {
+			commonRulesMap[*rule.RuleConfigurationName] = 0
+			latestRulesMap[*rule.RuleConfigurationName] = 0
+		}
+	}
+	for _, rule := range profilerRuleDesired {
+		if ackcompare.IsNotNil(rule.RuleConfigurationName) {
+			commonRulesMap[*rule.RuleConfigurationName] = 1
+		}
+	}
+	for _, val := range commonRulesMap {
+		// This means that there exists a rule in latest that is not present in desired
```

Comment:
This comment boils down to they tried/attempted to remove a profiler rule?

Reply:
correct